File: datastructure/Tree/leetcode_11.py
from Node import TreeNode

class Solution:
    sum = 0
    
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        self.cal(root)
        return self.sum
        
        
    def cal(self, root: TreeNode) -> int:
        if not root: return 0
        if not root.left and not root.right: return root.val
        
        ls = self.cal(root.left)
        rs = self.cal(root.right)
        
        self.sum += ls
        
        return 0

File: Leetcode(Swift)/Tree/Questions/Leetcode_11.swift
import Foundation

// Given the root of a binary tree, return the sum of all left leaves.

class Solution_11 {
    
    var totalSum = 0
    
    func sumOfLeftLeaves(_ root: TreeNode?) -> Int {
        guard root != nil else { return 0 }
        
        sum(of: root)
        
        return totalSum
    }
    
    private func sum(of node: TreeNode?) {
        guard let node = node else { return }
        
        if let left = node.left,
           left.left == nil &&
            left.right == nil {
            totalSum += node.left?.val ?? 0
        }
        
        sum(of: node.left)
        sum(of: node.right)
    }
}
